packages/web/src/hooks/useManagedAccountNotAllowedRedirect.ts

```diff
@@ -0,0 +1,51 @@
+import { useContext, useEffect } from 'react'
+
+import { useIsManagedAccount } from '@audius/common/hooks'
+
+import { ToastContext } from 'components/toast/ToastContext'
+import { FEED_PAGE } from 'utils/route'
+
+import { useNavigateToPage } from './useNavigateToPage'
+
+const messages = {
+  unauthorized: `You can't do that as a managed user`
+}
+
+/**
+ * Hook to redirect a managed account from accessing a particular route
+ * @param route route to redirect the user to
+ */
+export const useManagedAccountNotAllowedRedirect = (
+  route: string = FEED_PAGE
+) => {
+  const isManagedAccount = useIsManagedAccount()
+  const navigate = useNavigateToPage()
+  const { toast } = useContext(ToastContext)
+
+  useEffect(() => {
+    if (isManagedAccount) {
+      navigate(route)
+      toast(messages.unauthorized)
+    }
+  }, [isManagedAccount, navigate, route, toast])
+}
+
+/**
+ * Hook to prevent a managed account from some action (e.g. opening a modal)
+ * @param trigger condition that when true, fires the callback (e.g. modal is open)
+ * @param callback callback to fire when the action is performed (e.g. close the modal)
```

Comment:
I have this long-running internal debate about whether docstrings make it okay to use positional/unnamed boolean arguments (instead of a config object). A good IDE will show you the docs on hover. But reading the code elsewhere it's hard to understand what the arguments actually mean.

Reply:
probably worth making an obj 👍